test/extensions/filters/http/transformation/inja_transformer_replace_test.cc

```diff
@@ -0,0 +1,491 @@
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+
+  envoy::api::v2::filter::http::Extraction extractor;
+  extractor.mutable_body();
+  // Unless we are in `REPLACE_ALL` mode, we require regexes to match the entire target string
+  // because this only matches a substring, it should not be replaced
+  extractor.set_regex("body");
+  extractor.set_subgroup(0);
+  auto replacement_text = "BAZ";
+	extractor.mutable_replacement_text()->set_value(replacement_text);
+  extractor.set_mode(envoy::api::v2::filter::http::Extraction::SINGLE_REPLACE);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  std::string body("not json body");
+  GetBodyFunc bodyfunc = [&body]() -> const std::string & { return body; };
+  std::string res(Extractor(extractor).extract(callbacks, headers, bodyfunc));
+
+  EXPECT_EQ("", res);
+}
+
+TEST(Extraction, AttemptReplaceFromPartialMatchNonNilSubgroup) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+
+  envoy::api::v2::filter::http::Extraction extractor;
+  extractor.mutable_body();
+  // Unless we are in `REPLACE_ALL` mode, we require regexes to match the entire target string
+  // because this only matches a substring, it should not be replaced
+  // Note -- the subgroup in the regex is introduced here so that this config is not
+  // rejected when constructing the extractor
+  extractor.set_regex("(body)");
+  extractor.set_subgroup(1);
+  auto replacement_text = "BAZ";
+	extractor.mutable_replacement_text()->set_value(replacement_text);
+  extractor.set_mode(envoy::api::v2::filter::http::Extraction::SINGLE_REPLACE);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  std::string body("not json body");
+  GetBodyFunc bodyfunc = [&body]() -> const std::string & { return body; };
+  std::string res(Extractor(extractor).extract(callbacks, headers, bodyfunc));
+
+  EXPECT_EQ("", res);
+}
+
+TEST(Extraction, AttemptReplaceFromNoMatchNonNilSubgroup) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+
+  envoy::api::v2::filter::http::Extraction extractor;
+  extractor.mutable_body();
+  extractor.set_regex("(does not match)");
+  extractor.set_subgroup(1);
+  auto replacement_text = "BAZ";
+	extractor.mutable_replacement_text()->set_value(replacement_text);
+  extractor.set_mode(envoy::api::v2::filter::http::Extraction::SINGLE_REPLACE);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  std::string body("not json body");
+  GetBodyFunc bodyfunc = [&body]() -> const std::string & { return body; };
+  std::string res(Extractor(extractor).extract(callbacks, headers, bodyfunc));
+
+  EXPECT_EQ("", res);
+}
+
+TEST(Extraction, ReplaceFromFullLiteralMatch) {
+  Http::TestRequestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+
+  envoy::api::v2::filter::http::Extraction extractor;
+  extractor.mutable_body();
+  // We require regexes to match the entire target string
```

Comment:
this comment looks like it was copy-pasted - is it supposed to be here?

Reply:
Thanks, it was not supposed to be there. Resolved, w/ other test cleanup here: 202d18dfc7ca07ede7f41ed2f5abefcb9a2d0126